BUILD.md

```diff
@@ -0,0 +1,472 @@
+conduit tap deploy emojivoto/voting-svc
+```
+
+## Docker Compose
+
+To run all of the Go apps in a docker-compose environment:
+
+```bash
+docker-compose build
+docker-compose up -d
+
+# view dashboard
+open http://$DOCKER_IP:8084
+```
+
+If your system is configured to talk to a Kubernetes cluster, you can simulate
+traffic to the docker-compose environment:
+
+```bash
+# confirm you are connected to Kubernetes
+kubectl version
+
+# simulate traffic
+bin/go-run controller/script/simulate-proxy --kubeconfig ~/.kube/config --addr $DOCKER_IP:8086 --max-pods 10 --sleep 10ms
+```
+
+## Web
+
+This is a React app fronting a Go process. It uses webpack to bundle assets, and
+postcss to transform css.
+
+### First time setup
+
+Install [Yarn](https://yarnpkg.com) and use it to install dependencies:
+
+```bash
+brew install yarn
+cd web/app
+yarn
+```
+
+### Run web standalone
+
+```bash
+cd web/app
+yarn && yarn webpack
+cd ../..
+bin/go-run web
+```
+
+The web server will be running on `localhost:8084`.
+
+Note the `web` process depends on a `public-api` server, for which you have three
+options:
+
+#### 1. Connect to `public-api` locally
+
+```bash
+bin/go-run controller/cmd/public-api
+```
+
+#### 2. Connect to `public-api` in docker-compose
+
+Stop the web service, then run it locally and set the `--api-addr` flag to the
+address of the public API server that's running in your docker environment:
+
+```bash
+docker-compose stop web
+bin/go-run web --api-addr=$DOCKER_IP:8085
+```
+
+#### 3. Connect to `public-api` in Kubernetes
+
+If you are running the public API server in Kubernetes, forward `localhost:8085`
+to the Conduit controller pod:
+
+```bash
+POD_NAME=$(kubectl --namespace=conduit get po --selector=app=controller -o jsonpath='{.items[*].metadata.name}')
+kubectl -n conduit port-forward $POD_NAME 8085:8085
+```
+
+Then connect the local web process to the forwarded port:
+
+```bash
+bin/go-run web --api-addr=localhost:8085
+```
+
+### Webpack dev server
+
+To develop with a webpack dev server, start the server in a separate window:
+
+```bash
+cd web/app
+yarn webpack-dev-server
+```
+
+And then set the `--webpack-dev-server` flag when running the web server:
+
+```bash
+bin/go-run web --webpack-dev-server=http://localhost:8080
+```
+
+To add a JS dependency:
+
+```bash
+cd web/app
+yarn add [dep]
+```
+
+## Proxy
+
+To build and run the Rust proxy:
+
+```bash
+cargo build -p conduit-proxy
+CONDUIT_PROXY_LOG=trace \
+  CONDUIT_PROXY_PUBLIC_LISTENER=tcp://0:5432 \
+  CONDUIT_PROXY_PRIVATE_FORWARD=tcp://127.0.0.1:1234 \
+  CONDUIT_PROXY_CONTROL_URL=tcp://127.0.0.1:8086 \
+  target/debug/conduit-proxy
+```
+
+To connect to a live `proxy-api` at `localhost:8086`:
+
+```bash
+bin/go-run controller/cmd/proxy-api
+```
+
+# Testing
+
+## Go
+
+```bash
+dep ensure && dep prune
+go test ./...
+go vet ./...
+```
+
+## JavaScript
+
+```bash
+cd web/app
+yarn && yarn webpack
+yarn karma start --single-run
+```
+
+## Rust
+
+```bash
+cargo check
```

Comment:
this is essentially the same as running `cargo build` without producing a binary; it just invokes the analysis portion of the  compiler and skips codegen. useful if you want to know if your changes will compile, but not sure if it's really "testing"...

Reply:
sounds like `cargo test` is a superset. i'm going to leave this here for now so that it matches `.travis.yml`. i've filed https://github.com/runconduit/conduit/issues/149 to follow up.